### PR TITLE
RIG Chem Dispenser rework.

### PR DESCRIPTION
**Pull request by MLGTASTICa**
## About The Pull Request
Fully reworks the rig chemical injectors , most notable changes
1)You can now insert/remove beakers
2)You can adjust the injection amount (and saboutage if you set it to 0)
3)Combat injectors now have hyperzine , nutriment , tricordazine and tramadol
4)Medical injectors now have dylovene , bicardine , kelotane , spaceacilin and inaprovaline
5)Modular injectors no longer start full , only their subtypes (RND printing injectors won't start full now)
6)Normal injector can fit 5 beakers , subtypes can fit 6
7)Uplink modular injector price changed from 6 to 3 tc(no longer starts full)

## Why It's Good For The Game
They suck in general , and were hard to refill , and use in general , now this  makes them more flexible for players and to their needs , especially paramedics which can finnaly fill theirs with upgraded chems.
## Changelog
:cl:
balance: RIG Combat dispenser now has hyperzine , nutriment , tricord and tramadol
balance :RIG medical dispenser added , with dylovene , bicardine , kelotane spaceacilin and inaprovaline
balance: You can now add and remove beakers from modular injectors
balance: Printed modular injector(chem dispensers) no longer start filled
balance: Uplink injector price changed from 6 to 3 tc as it no longer starts full.
/:cl:

### CODE REVIEW

File: code/modules/clothing/spacesuits/rig/modules/utility.dm [hunk 1]
```diff
@@ -113,8 +113,158 @@
...
+			charge_dat.product_type = ref(beaker)
+			charge_dat.charges      = beaker.reagents.total_volume
+
+			if(!charge_selected) charge_selected = charge_dat.short_name
+			processed_charges[charge_dat.short_name] = charge_dat
+
+		charges = processed_charges
+
+
+/obj/item/rig_module/modular_injector/accepts_item(obj/item/reagent_containers/item, mob/living/user,userless = FALSE)
```
Comment (hyperioo) [inline code/modules/clothing/spacesuits/rig/modules/utility.dm:169]:
```suggestion
/obj/item/rig_module/modular_injector/accepts_item(obj/item/reagent_containers/item, mob/living/user, userless = FALSE)
```

File: code/modules/clothing/spacesuits/rig/modules/utility.dm [hunk 2]
```diff
@@ -113,8 +113,158 @@
...
+	if(!istype(item))
+		return FALSE
+	if(beakers.len == max_beakers)
+		to_chat(user, "\The [src] has all its beaker slots filled , remove one of them!")
```
Comment (hyperioo) [inline code/modules/clothing/spacesuits/rig/modules/utility.dm:173]:
```suggestion
		to_chat(user, "\The [src] has all its beaker slots filled, remove one of them!")
```

File: code/modules/clothing/spacesuits/rig/modules/utility.dm [hunk 3]
```diff
@@ -113,8 +113,158 @@
...
+	initial_beakers = list(
+		list(/obj/item/reagent_containers/glass/beaker/large, "hyperzine", 60),
+		list(/obj/item/reagent_containers/glass/beaker/large, "tramadol", 60),
+		list(/obj/item/reagent_containers/glass/beaker/large, "nutriment", 60),
+		list(/obj/item/reagent_containers/glass/beaker/large, "tricordrazine", 60)
+	)
+
+/obj/item/rig_module/modular_injector/medical
+	name = "mounted medical injector"
+	desc = "A specialized system for inserting chemicals to pacients"
```
Comment (hyperioo) [inline code/modules/clothing/spacesuits/rig/modules/utility.dm:257]:
```suggestion
	desc = "A specialized system for inserting chemicals to patients"
```

**Review by hyperioo — CHANGES_REQUESTED** [review]

File: code/modules/clothing/spacesuits/rig/modules/utility.dm (hunk 3 above)
Comment (MLGTASTICa) [reply code/modules/clothing/spacesuits/rig/modules/utility.dm:257]: ? its pacients.

**Review by hyperioo — APPROVED** [review]

**Comment by hyperioo** [comment]
Ah fuck I did not see you did not commited the patients... Well...

File: code/modules/clothing/spacesuits/rig/modules/utility.dm (hunk 3 above)
Comment (hyperioo) [reply code/modules/clothing/spacesuits/rig/modules/utility.dm:257]: image

**Pull request merged**